model/Core/Classes/Build.md

```diff
@@ -0,0 +1,74 @@
+SPDX-License-Identifier: Community-Spec-1.0
+
+# Build
+
+## Summary
+
+Class that describes a build of software/artifacts.
+
+## Description
+
+A build is a representation of the process in which a piece of software or
+artifact is built. It encapsulates information related to a build process and
+provides an element from which relationships can be created to relate the build
+to the inputs/outputs, and entities related to the build (e.g. builders,
+identities, etc.).
+
+Definitions of "BuildType", "ConfigSource", "Parameters" and "Environment" follow
+those defined in [SLSA provenance](https://slsa.dev/provenance/v0.2).
+
+<!--
+The build will include relationships such as:
+
+<Build> BUILT_BY <SPDX_Element> 
+<Build> BUILD_INITIATED_BY <Subject/Actor>
+<Package> BUILD_INPUT_OF <Build>
+<Package> BUILD_OUTPUT_OF <Build>
+
+For BUILT_BY:
+At the moment it can point to an artifact to signify a builder, but can point
+to a "Provider/Service" element in the future which can encapsulate more
+details of the builder if needed.
+-->
+
+## Metadata
+
+- name: Build
+- SubclassOf: Element
+- Instantiability: Concrete
+
+## Properties
+
+- buildType
+  - type: xsd:string
+  - minCount: 0
+- configSoureEntrypoint
+  - type: xsd:string
+  - minCount: 0
+- configSoureUri
+  - type: xsd:string
+  - minCount: 0
+- configSoureDigest
+  - type: xsd:string
+  - minCount: 0
+- parameters
+  - type: xsd:map<string>string
+  - minCount: 0
+- buildInvocationId
```

Comment:
Is this intended to be a unique identifier for the build instance? If so, since `Build` inherits from `Element` could this be stored in the `ExternalIdentifiers`?

Reply:
From my understanding of the discussions in recent Builds SIG meetings, you're spot on, @iamwillbar. I would also agree that `ExternalIdentifiers` is the right place to put this, given that the cardinality of the property is listed as 0-1 (that is, it's an optional field). However, this would mean that we would need to add new ExternalIdentifier types to support different build systems' ID schemes.

Reply:
From my understanding of the discussions in recent Builds SIG meetings, you're spot on, @iamwillbar. I would also agree that ExternalIdentifiers is the right place to put this, given that the cardinality of the property is listed as 0-1 (that is, it's an optional field). However, this would mean that we would need to add new ExternalIdentifier types to support different build systems' ID schemes.